### PR TITLE
develop: refactor Python dev guide

### PR DESCRIPTION
**Pull request by wd0517**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by sykp241095 — APPROVED** [review]

File: develop/dev-guide-sample-application-python-django.md [hunk 1]
```diff
@@ -1,783 +1,316 @@
 ---
-title: 使用 Django 构建 TiDB 应用程序
-summary: 给出一个 Django 构建 TiDB 应用程序示例。
-aliases: ['/zh/tidb/dev/dev-guide-sample-application-django']
+title: 使用 Django 连接到 TiDB
+summary: 了解如何使用 Django 连接到 TiDB。本文提供了使用 Django 与 TiDB 交互的 Python 示例代码片段。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:3]:
```suggestion
summary: 了解如何使用 Django 连接到 TiDB。本文提供了使用 Django 与 TiDB 交互的 Python 示例代码片段。
aliases: ['/zh/tidb/dev/dev-guide-sample-application-django']
```

File: develop/dev-guide-sample-application-python-django.md [hunk 2]
```diff
@@ -1,783 +1,316 @@
...
 ---
 
-<!-- markdownlint-disable MD029 -->
+# 使用 Django 连接到 TiDB
 
-# 使用 Django 构建 TiDB 应用程序
+TiDB 是一个兼容 MySQL 的数据库。[Django](https://www.djangoproject.com/) 是一个非常流行的 Python Web 框架，它内部实现了一个强大的 ORM(Object Relational Mapper) 系统。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:8]:
```suggestion
TiDB 是一个兼容 MySQL 的数据库。[Django](https://www.djangoproject.com/) 为当前流行的 Python Web 框架之一，它内部实现了一个强大的 ORM (Object Relational Mapper) 系统。
```

File: develop/dev-guide-sample-application-python-django.md [hunk 3]
```diff
@@ -1,783 +1,316 @@
...
-> **建议：**
->
-> 如果你希望得到一个与本示例相同依赖的空白程序，而无需示例代码，可参考[创建相同依赖空白程序（可选）](#创建相同依赖空白程序可选)一节。
+如果你在安装 mysqlclient 时遇到问题，请参考 [mysqlclient 官方文档](https://github.com/PyMySQL/mysqlclient#install)。
 
-请下载或克隆示例代码库 [pingcap-inc/tidb-example-python](https://github.com/pingcap-inc/tidb-example-python)，并进入到目录 `django_example` 中。
+#### `django-tidb` 是什么？
 
-## 第 4 步：运行应用程序
+django-tidb 解决了 TiDB 和 Django 之间的兼容性问题。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:53]:
```suggestion
`django-tidb` 是为 Django 开发的 TiDB 方言，它解决了 TiDB 和 Django 之间的兼容性问题。
```

File: develop/dev-guide-sample-application-python-django.md [hunk 4]
```diff
@@ -1,783 +1,316 @@
...
 
-接下来运行应用程序代码，将会生成一个 Web 应用程序。你可以使用 `python manage.py migrate` 命令，要求 Django 在数据库 `django` 中创建一个表 `player`。如果你向应用程序的 Restful API 发送请求，这些请求将会在 TiDB 集群上运行[数据库事务](/develop/dev-guide-transaction-overview.md)。
+安装 django-tidb 的时候，你需要选择与你的 Django 版本对应的版本，例如，如果你使用的是 `django==4.2.*`，你需要安装 `django-tidb==4.2.*`，最小版本号不需要一致，使用各自最新的最小版本即可。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:55]:
```suggestion
安装 `django-tidb` 时，请选择与你的 Django 版本匹配的版本。例如，如果你使用的是 `django==4.2.*`，则应安装 `django-tidb==4.2.*`，其中 minor 版本号不需要完全相同。建议使用最新的 minor 版本。
```

File: develop/dev-guide-sample-application-python-django.md [hunk 5]
```diff
@@ -1,783 +1,316 @@
...
 
-</div>
-
-</SimpleTab>
-
-## 实现细节
+### 第 5 步：运行示例应用程序
 
-本小节介绍示例应用程序项目中的组件。
+1. 运行以下命令，使用开发模式启动示例应用程序：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:179]:
```suggestion
1. 在开发模式下运行示例应用程序：
```

File: develop/dev-guide-sample-application-python-django.md [hunk 6]
```diff
@@ -1,783 +1,316 @@
...
-│   ├── migrations
-│   │   ├── 0001_initial.py
-│   │   └── __init__.py
-│   ├── models.py
-│   ├── tests.py
-│   ├── urls.py
-│   └── views.py
-└── manage.py
-```
+    默认的端口号为 8000，如果你想要修改端口号，可以在命令后面添加端口号，例如：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:185]:
```suggestion
    应用程序默认在 `8000` 端口上运行。如果你想要使用其他端口号，可以在命令后添加端口号，例如：
```

File: develop/dev-guide-sample-application-python-django.md [hunk 7]
```diff
@@ -1,783 +1,316 @@
...
+2. 打开浏览器，在地址栏输入 `http://localhost:8000/`，访问示例应用程序，你可以进行以下操作：
 
-    - `settings.py` 声明了项目的配置，如数据库地址、密码、使用的数据库方言等。
-    - `urls.py` 配置了项目的根路由。
+    - 创建一个新的 Player
+    - 批量创建 Player
+    - 查看所有的 Player
+    - 更新 Player
+    - 删除 Player
+    - 在两个 Player 之间进行交易
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:198]:
```suggestion
    - 在两个 Player 之间交易物品
```

File: develop/dev-guide-sample-application-python-django.md [hunk 8]
```diff
@@ -1,783 +1,316 @@
...
-> **注意：**
->
-> 由于 Django 的设计采用了可插拔模式，因此，你需要在创建应用后，在项目中进行注册。在本示例中，注册过程就是在 `example_project/settings.py` 文件中，在 `INSTALLED_APPS` 对象内添加 `'player.apps.PlayerConfig'` 条目。你可以参考示例代码 [`settings.py`](https://github.com/pingcap-inc/tidb-example-python/blob/main/django_example/example_project/settings.py#L33-L41) 以获得更多信息。
+完整代码及其运行方式，见代码仓库 [tidb-samples/tidb-python-django-quickstart](https://github.com/tidb-samples/tidb-python-django-quickstart)。
 
-### 项目配置
+### 配置数据库连接
 
-本节将简要介绍 `example_project` 包内 `settings.py` 的重要配置。这个文件包含了 Django 项目的配置，声明了项目包含的应用、中间件、连接的数据库等信息。你可以通过[创建相同依赖空白程序](#创建相同依赖空白程序可选)这一节来了解此配置文件的生成流程，也可直接在项目中使用 `settings.py` 文件。关于 Django 配置的更多信息，参考 [Django 配置](https://docs.djangoproject.com/zh-hans/3.2/topics/settings/)文档。
+打开 `sample_project/settings.py` 文件，添加以下配置：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:208]:
```suggestion
在 `sample_project/settings.py` 文件中添加以下配置：
```

File: develop/dev-guide-sample-application-python-mysql-connector.md [hunk 9]
```diff
@@ -1,283 +1,269 @@
 ---
-title: TiDB 和 MySQL Connector/Python 的简单 CRUD 应用程序
-summary: 给出一个 TiDB 和 MySQL Connector/Python 的简单 CRUD 应用程序示例。
-aliases: ['/zh/tidb/dev/dev-guide-sample-application-python']
+title: 使用 MySQL Connector/Python 连接到 TiDB
+summary: 了解如何使用 MySQL Connector/Python 连接到 TiDB。本文提供了使用 MySQL Connector/Python 与 TiDB 交互的 Python 示例代码片段。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysql-connector.md:3]:
```suggestion
summary: 了解如何使用 MySQL Connector/Python 连接到 TiDB。本文提供了使用 MySQL Connector/Python 与 TiDB 交互的 Python 示例代码片段。
aliases: ['/zh/tidb/dev/dev-guide-sample-application-python']
```

File: develop/dev-guide-sample-application-python-mysql-connector.md [hunk 10]
```diff
@@ -1,283 +1,269 @@
...
 
-当以后需要多次运行脚本时，请在每次运行前先依照[表初始化](#第-3-步第-1-部分表初始化)一节再次进行表初始化。
+Python 驱动程序提供对数据库的底层访问，但要求开发者：
 
-```bash
-python3 mysql_connector_python_example.py
-```
+- 手动建立和释放数据库连接
+- 手动管理数据库事务
+- 手动将数据行（在 `mysql-connector-python` 中表示为元组(tuple)或者字典(dict)）映射为数据对象
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysql-connector.md:254]:
```suggestion
- 手动将数据行（在 `mysql-connector-python` 中表示为元组 (tuple) 或者字典 (dict)）映射为数据对象
```

File: develop/dev-guide-sample-application-python-pymysql.md [hunk 11]
```diff
@@ -1,277 +1,274 @@
...
 
-当以后需要多次运行脚本时，请在每次运行前先依照[表初始化](#第-3-步第-1-部分表初始化)一节再次进行表初始化。
+Python 驱动程序提供对数据库的底层访问，但要求开发者：
 
-```bash
-python3 pymysql_example.py
-```
+- 手动建立和释放数据库连接
+- 手动管理数据库事务
+- 手动将数据行（在 pymysql 中表示为元组 (tuple)或者字典(dict)）映射为数据对象
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-pymysql.md:259]:
```suggestion
- 手动将数据行（在 pymysql 中表示为元组 (tuple) 或者字典 (dict)）映射为数据对象
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 12]
```diff
@@ -1,238 +1,283 @@
...
-from sqlalchemy.orm import declarative_base, sessionmaker
+```bash
+pip install -r requirements.txt
+```
 
-engine = create_engine('mysql://root:@127.0.0.1:4000/test')
-Base = declarative_base()
-Base.metadata.create_all(engine)
-Session = sessionmaker(bind=engine)
+#### 为什么需要安装 PyMySQL？
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:49]:
```suggestion
#### 为什么安装 PyMySQL？
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 13]
```diff
@@ -1,238 +1,283 @@
...
 
+SQLAlchemy 是一个 ORM 库，支持多种数据库。它是对数据库的高层抽象，可以帮助开发者以更面向对象的方式编写 SQL 语句。但 SQLAlchemy 并不提供数据库驱动，因此需要单独安装用于连接 TiDB 的驱动。本示例项目使用 PyMySQL 作为数据库驱动。PyMySQL 是一个纯 Python 实现的 MySQL 客户端库，兼容 TiDB，可以在所有平台上轻松安装。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:51]:
```suggestion
SQLAlchemy 是一个支持多种数据库的 ORM 库。它是对数据库的高层抽象，可以帮助开发者以更面向对象的方式编写 SQL 语句。但 SQLAlchemy 并不提供数据库驱动，因此需要单独安装用于连接 TiDB 的驱动。本示例项目使用 PyMySQL 作为数据库驱动。PyMySQL 是一个与 TiDB 兼容的纯 Python 实现的 MySQL 客户端库，并可以在所有平台上安装。
```

File: develop/dev-guide-sample-application-python-sqlalchemy.md [hunk 14]
```diff
@@ -1,238 +1,283 @@
...
 
-class Player(Base):
-    __tablename__ = "player"
+你也可以使用其他数据库驱动，例如 [mysqlclient](https://github.com/PyMySQL/mysqlclient) 以及 [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en/)，但是它们不是纯 Python 库，需要安装对应的 C/C++ 编译器和 MySQL 客户端库来编译。更多信息，参考 [SQLAlchemy 官方文档](https://docs.sqlalchemy.org/en/20/core/engines.html#mysql)。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-sqlalchemy.md:53]:
```suggestion
你也可以使用其他数据库驱动，例如 [mysqlclient](https://github.com/PyMySQL/mysqlclient) 以及 [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en/)。但是它们不是纯 Python 库，需要安装对应的 C/C++ 编译器和 MySQL 客户端库进行编译。更多信息，参考 [SQLAlchemy 官方文档](https://docs.sqlalchemy.org/en/20/core/engines.html#mysql)。
```

File: develop/dev-guide-sample-application-python-django.md
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-django.md:1]: Please also update the link text in develop/dev-guide-choose-driver-or-orm.md and develop/dev-guide-insert-data.md

https://github.com/pingcap/docs/pull/14638/

File: develop/dev-guide-sample-application-python-django.md (hunk 3 above)
Comment (sykp241095) [reply develop/dev-guide-sample-application-python-django.md:54]: @wd0517 PTAL is it suitable to use "方言"?

File: develop/dev-guide-choose-driver-or-orm.md [hunk 15]
```diff
@@ -243,7 +243,7 @@ go get -u gorm.io/driver/mysql
 
 按照 [PyMySQL 文档](https://pypi.org/project/PyMySQL/)中的说明下载并配置驱动程序即可使用。建议使用 **1.0.2** 及以上版本。
 
-使用 PyMySQL 构建 TiDB 应用程序的例子，可参阅 [TiDB 和 PyMySQL 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-python-pymysql.md)。
+使用 PyMySQL 构建 TiDB 应用程序的例子，可参阅 [使用 PyMySQL 连接到 TiDB](/develop/dev-guide-sample-application-python-pymysql.md)。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:246]:
```suggestion
使用 PyMySQL 构建 TiDB 应用程序的例子，可参阅[使用 PyMySQL 连接到 TiDB](/develop/dev-guide-sample-application-python-pymysql.md)。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 16]
```diff
@@ -252,7 +252,7 @@ go get -u gorm.io/driver/mysql
 
 按照 [mysqlclient 文档](https://pypi.org/project/mysqlclient/)中的说明下载并配置驱动程序即可使用。建议使用 **2.1.1** 及以上版本。
 
-使用 mysqlclient 构建 TiDB 应用程序的例子，可参阅 [TiDB 和 mysqlclient 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-python-mysqlclient.md)。
+使用 mysqlclient 构建 TiDB 应用程序的例子，可参阅 [使用 mysqlclient 连接到 TiDB](/develop/dev-guide-sample-application-python-mysqlclient.md)。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:255]:
```suggestion
使用 mysqlclient 构建 TiDB 应用程序的例子，可参阅[使用 mysqlclient 连接到 TiDB](/develop/dev-guide-sample-application-python-mysqlclient.md)。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 17]
```diff
@@ -261,7 +261,7 @@ go get -u gorm.io/driver/mysql
 
 按照 [MySQL Connector/Python 文档](https://dev.mysql.com/doc/connector-python/en/connector-python-installation-binary.html)中的说明下载并配置驱动程序即可使用。建议使用 **8.0.31** 及以上版本。
 
-使用 MySQL Connector/Python 构建 TiDB 应用程序的例子，可参阅 [TiDB 和 MySQL Connector/Python 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-python-mysql-connector.md)。
+使用 MySQL Connector/Python 构建 TiDB 应用程序的例子，可参阅 [使用 MySQL Connector/Python 连接到 TiDB](/develop/dev-guide-sample-application-python-mysql-connector.md)。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:264]:
```suggestion
使用 MySQL Connector/Python 构建 TiDB 应用程序的例子，可参阅[使用 MySQL Connector/Python 连接到 TiDB](/develop/dev-guide-sample-application-python-mysql-connector.md)。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 18]
```diff
@@ -286,7 +286,7 @@ go get -u gorm.io/driver/mysql
 
 [SQLAlchemy](https://www.sqlalchemy.org/) 是一个流行的 Python 的 ORM 框架，你可以使用 `pip install SQLAlchemy==1.4.44` 获取你的应用程序的所有依赖项。建议使用 **1.4.44** 及以上版本。
 
-使用 SQLAlchemy 构建 TiDB 应用程序的例子，可参阅 [TiDB 和 SQLAlchemy 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-python-sqlalchemy.md)。
+使用 SQLAlchemy 构建 TiDB 应用程序的例子，可参阅 [使用 SQLAlchemy 连接到 TiDB](/develop/dev-guide-sample-application-python-sqlalchemy.md)。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:289]:
```suggestion
使用 SQLAlchemy 构建 TiDB 应用程序的例子，可参阅[使用 SQLAlchemy 连接到 TiDB](/develop/dev-guide-sample-application-python-sqlalchemy.md)。
```

File: develop/dev-guide-choose-driver-or-orm.md [hunk 19]
```diff
@@ -296,7 +296,7 @@ go get -u gorm.io/driver/mysql
 
 [peewee](http://docs.peewee-orm.com/en/latest/) 是一个流行的 Python 的 ORM 框架，你可以使用 `pip install peewee==3.15.4` 获取你的应用程序的所有依赖项。建议使用 **3.15.4** 及以上版本。
 
-使用 peewee 构建 TiDB 应用程序的例子，可参阅 [TiDB 和 peewee 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-python-peewee.md)。
+使用 peewee 构建 TiDB 应用程序的例子，可参阅 [使用 peewee 连接到 TiDB](/develop/dev-guide-sample-application-python-peewee.md)。
```
Comment (Oreoxmt) [inline develop/dev-guide-choose-driver-or-orm.md:299]:
```suggestion
使用 peewee 构建 TiDB 应用程序的例子，可参阅[使用 peewee 连接到 TiDB](/develop/dev-guide-sample-application-python-peewee.md)。
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**